infra/main.bicep

```diff
@@ -0,0 +1,314 @@
+/*                                 VARIABLES                                  */
+/******************************************************************************/
+
+var abbreviations = loadJsonContent('./abbreviations.json')
+
+// tags that should be applied to all resources.
+var tags = {
+  // Tag all resources with the environment name.
+  'azd-env-name': environmentName
+}
+
+/******************************* Resource Names *******************************/
+
+
+// Generate a unique token to be used in naming resources.
+// Remove linter suppression after using.
+#disable-next-line no-unused-vars
+var resourceToken = toLower(uniqueString(subscription().id, environmentName, location))
+
+@description('Name of the environment with only alphanumeric characters. Used for resource names that require alphanumeric characters only.')
+var alphaNumericEnvironmentName = replace(replace(environmentName, '-', ''), ' ', '')
+
+var _containerAppsEnvironmentName = !empty(containerAppsEnvironmentName) ? containerAppsEnvironmentName : take('${abbreviations.appManagedEnvironments}${environmentName}', 60)
+var _containerRegistryName = !empty(containerRegistryName) ? containerRegistryName : take('${abbreviations.containerRegistryRegistries}${take(alphaNumericEnvironmentName, 35)}${resourceToken}', 50)
+var _logAnalyticsName = !empty(logAnalyticsWorkspaceName) ? logAnalyticsWorkspaceName : take('${abbreviations.operationalInsightsWorkspaces}${environmentName}', 63)
+var _applicationInsightsName = !empty(applicationInsightsName) ? applicationInsightsName : take('${abbreviations.insightsComponents}${environmentName}', 255)
+var _applicationInsightsDashboardName = !empty(applicationInsightsDashboardName) ? applicationInsightsDashboardName : take('${abbreviations.portalDashboards}${environmentName}', 160)
+var _quarkusContainerAppName = !empty(quarkusContainerAppName) ? quarkusContainerAppName : take('${abbreviations.appContainerApps}quarkus-${environmentName}', 32)
+var _springBootContainerAppName = !empty(springBootContainerAppName) ? springBootContainerAppName : take('${abbreviations.appContainerApps}spring-boot-${environmentName}', 32)
+var _postgresFlexibleServerName = !empty(postgresFlexibleServerName) ? postgresFlexibleServerName : take(toLower('${abbreviations.dBforPostgreSQLServers}${take(environmentName, 44)}-${resourceToken}'), 63)
+var _keyVaultName = !empty(keyVaultName) ? keyVaultName : take('${abbreviations.keyVaultVaults}${take(alphaNumericEnvironmentName, 8)}${resourceToken}', 24)
+var _keyVaultSecrets = [
+  {
+    name: 'postgres-admin-password'
+    value: postgresAdminPassword
+  }
+]
+// Name of the service defined in azure.yaml
+// A tag named azd-service-name with this value should be applied to the service host resource, such as:
+//   Microsoft.Web/sites for appservice, function
+// Example usage:
+//   tags: union(tags, { 'azd-service-name': apiServiceName })
+#disable-next-line no-unused-vars
```

Comment:
Can you use these vars?

Reply:
Thanks for noticing this one, service tagging was missing.